app/src/main/java/org/mozilla/fenix/HomeActivity.kt

```diff
@@ -263,6 +269,31 @@ open class HomeActivity : LocaleAwareAppCompatActivity(), NavHostActivity {
         components.appStartupTelemetry.onHomeActivityOnRestart(rootContainer)
     }
 
+    private fun createInflaterFactory() {
+        factory = object : LayoutInflater.Factory2 {
+            override fun onCreateView(parent: View?, name: String, context: Context, attrs: AttributeSet): View? =
+                    when (name) {
+                        EngineView::class.java.name -> components.core.engine.createView(context, attrs).apply {
+                            selectionActionDelegate = DefaultSelectionActionDelegate(
+                                    BrowserStoreSearchAdapter(
+                                            components.core.store,
```

Comment:
This is an unreasonable amount of indentation: this is 8 levels and I think ~3-4 is the usual comfortable amount. Can we declare this inflater class in a separate file to lower it? Also the indentation changed from the original PR – you have some double-indentations. Can you fix that?

In general, anonymous classes also worry me as they're harder to understand because they have access to many more references.